### ISSUE TITLE
Fails to build on Ubuntu 20.04

### ISSUE THREAD

**Issue by cc-michaelu** [curl/curl#8624]
### I did this
I cloned the curl repository. CD'd into curl directory, created a build directory within, cd'd to the build directory, ran `cmake ..` to generate makefiles. Ran make.

### I expected the following
I expected that the curl binary would be generated. Instead I receive the following error message:
```
[main] Building folder: curl 
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/michael_uman/gitroot/curl/build --config Debug --target all --
[build] [14/203   0% :: 0.069] Building C object lib/CMakeFiles/libcurl.dir/c-hyper.c.o
[build] [15/203   0% :: 0.080] Building C object lib/CMakeFiles/libcurl.dir/amigaos.c.o
[build] [16/203   1% :: 0.081] Building C object lib/CMakeFiles/libcurl.dir/curl_ctype.c.o
[build] [17/203   1% :: 0.097] Building C object lib/CMakeFiles/libcurl.dir/curl_addrinfo.c.o
[build] [18/203   2% :: 0.105] Building C object lib/CMakeFiles/libcurl.dir/asyn-ares.c.o
[build] [19/203   2% :: 0.121] Building C object lib/CMakeFiles/libcurl.dir/curl_des.c.o
[build] [20/203   3% :: 0.140] Building C object lib/CMakeFiles/libcurl.dir/altsvc.c.o
[build] [21/203   3% :: 0.158] Building C object lib/CMakeFiles/libcurl.dir/asyn-thread.c.o
[build] [22/203   4% :: 0.159] Building C object lib/CMakeFiles/libcurl.dir/base64.c.o
[build] [23/203   4% :: 0.159] Building C object lib/CMakeFiles/libcurl.dir/bufref.c.o
[build] [24/203   5% :: 0.159] Building C object lib/CMakeFiles/libcurl.dir/content_encoding.c.o
[build] [25/203   5% :: 0.160] Building C object lib/CMakeFiles/libcurl.dir/curl_endian.c.o
[build] [26/203   6% :: 0.161] Building C object lib/CMakeFiles/libcurl.dir/curl_get_line.c.o
[build] [27/203   6% :: 0.170] Building C object lib/CMakeFiles/libcurl.dir/conncache.c.o
[build] [28/203   7% :: 0.171] Building C object lib/CMakeFiles/libcurl.dir/curl_gethostname.c.o
[build] [29/203   7% :: 0.181] Building C object lib/CMakeFiles/libcurl.dir/curl_fnmatch.c.o
[build] [30/203   8% :: 0.194] Building C object lib/CMakeFiles/libcurl.dir/connect.c.o
[build] [31/203   8% :: 0.196] Building C object lib/CMakeFiles/libcurl.dir/curl_gssapi.c.o
[build] [32/203   9% :: 0.203] Building C object lib/CMakeFiles/libcurl.dir/cookie.c.o
[build] [33/203   9% :: 0.219] Building C object lib/CMakeFiles/libcurl.dir/curl_memrchr.c.o
[build] [34/203  10% :: 0.230] Building C object lib/CMakeFiles/libcurl.dir/curl_ntlm_wb.c.o
[build] [35/203  10% :: 0.232] Building C object lib/CMakeFiles/libcurl.dir/curl_multibyte.c.o
[build] [36/203  11% :: 0.233] Building C object lib/CMakeFiles/libcurl.dir/curl_path.c.o
[build] [37/203  11% :: 0.249] Building C object lib/CMakeFiles/libcurl.dir/curl_sspi.c.o
[build] [38/203  12% :: 0.260] Building C object lib/CMakeFiles/libcurl.dir/curl_rtmp.c.o
[build] [39/203  12% :: 0.263] Building C object lib/CMakeFiles/libcurl.dir/curl_range.c.o
[build] [40/203  13% :: 0.267] Building C object lib/CMakeFiles/libcurl.dir/curl_threads.c.o
[build] [41/203  13% :: 0.294] Building C object lib/CMakeFiles/libcurl.dir/dotdot.c.o
[build] [42/203  14% :: 0.316] Building C object lib/CMakeFiles/libcurl.dir/dynbuf.c.o
[build] [43/203  14% :: 0.334] Building C object lib/CMakeFiles/libcurl.dir/curl_sasl.c.o
[build] [44/203  15% :: 0.335] Building C object lib/CMakeFiles/libcurl.dir/easygetopt.c.o
[build] [45/203  15% :: 0.336] Building C object lib/CMakeFiles/libcurl.dir/dict.c.o
[build] [46/203  16% :: 0.346] Building C object lib/CMakeFiles/libcurl.dir/easyoptions.c.o
[build] [47/203  16% :: 0.366] Building C object lib/CMakeFiles/libcurl.dir/curl_ntlm_core.c.o
[build] [48/203  17% :: 0.386] Building C object lib/CMakeFiles/libcurl.dir/escape.c.o
[build] [49/203  17% :: 0.410] Building C object lib/CMakeFiles/libcurl.dir/fileinfo.c.o
[build] [50/203  18% :: 0.415] Building C object lib/CMakeFiles/libcurl.dir/getenv.c.o
[build] [51/203  18% :: 0.426] Building C object lib/CMakeFiles/libcurl.dir/file.c.o
[build] [52/203  19% :: 0.435] Building C object lib/CMakeFiles/libcurl.dir/easy.c.o
[build] [53/203  19% :: 0.435] Building C object lib/CMakeFiles/libcurl.dir/formdata.c.o
[build] [54/203  20% :: 0.445] Building C object lib/CMakeFiles/libcurl.dir/doh.c.o
[build] [55/203  20% :: 0.458] Building C object lib/CMakeFiles/libcurl.dir/gopher.c.o
[build] [56/203  21% :: 0.459] Building C object lib/CMakeFiles/libcurl.dir/hash.c.o
[build] [57/203  21% :: 0.487] Building C object lib/CMakeFiles/libcurl.dir/getinfo.c.o
[build] [58/203  22% :: 0.493] Building C object lib/CMakeFiles/libcurl.dir/headers.c.o
[build] [59/203  22% :: 0.509] Building C object lib/CMakeFiles/libcurl.dir/hostip4.c.o
[build] [60/203  23% :: 0.511] Building C object lib/CMakeFiles/libcurl.dir/ftplistparser.c.o
[build] [61/203  23% :: 0.512] Building C object lib/CMakeFiles/libcurl.dir/h2h3.c.o
[build] [62/203  24% :: 0.523] Building C object lib/CMakeFiles/libcurl.dir/hmac.c.o
[build] [63/203  24% :: 0.523] Building C object lib/CMakeFiles/libcurl.dir/hostasyn.c.o
[build] [64/203  25% :: 0.524] Building C object lib/CMakeFiles/libcurl.dir/hostsyn.c.o
[build] [65/203  25% :: 0.536] Building C object lib/CMakeFiles/libcurl.dir/hostip6.c.o
[build] [66/203  26% :: 0.559] Building C object lib/CMakeFiles/libcurl.dir/http2.c.o
[build] [67/203  26% :: 0.580] Building C object lib/CMakeFiles/libcurl.dir/http_negotiate.c.o
[build] [68/203  27% :: 0.603] Building C object lib/CMakeFiles/libcurl.dir/hsts.c.o
[build] [69/203  27% :: 0.613] Building C object lib/CMakeFiles/libcurl.dir/http_chunks.c.o
[build] [70/203  28% :: 0.618] Building C object lib/CMakeFiles/libcurl.dir/http_digest.c.o
[build] [71/203  28% :: 0.634] Building C object lib/CMakeFiles/libcurl.dir/idn_win32.c.o
[build] [72/203  29% :: 0.639] Building C object lib/CMakeFiles/libcurl.dir/hostip.c.o
[build] [73/203  29% :: 0.649] Building C object lib/CMakeFiles/libcurl.dir/if2ip.c.o
[build] [74/203  30% :: 0.652] Building C object lib/CMakeFiles/libcurl.dir/http_aws_sigv4.c.o
[build] [75/203  30% :: 0.654] Building C object lib/CMakeFiles/libcurl.dir/ftp.c.o
[build] [76/203  31% :: 0.659] Building C object lib/CMakeFiles/libcurl.dir/inet_pton.c.o
[build] [77/203  31% :: 0.661] Building C object lib/CMakeFiles/libcurl.dir/inet_ntop.c.o
[build] [78/203  32% :: 0.678] Building C object lib/CMakeFiles/libcurl.dir/http_ntlm.c.o
[build] [79/203  32% :: 0.685] Building C object lib/CMakeFiles/libcurl.dir/http_proxy.c.o
[build] [80/203  33% :: 0.689] Building C object lib/CMakeFiles/libcurl.dir/ldap.c.o
[build] [81/203  33% :: 0.703] Building C object lib/CMakeFiles/libcurl.dir/krb5.c.o
[build] [82/203  33% :: 0.711] Building C object lib/CMakeFiles/libcurl.dir/md4.c.o
[build] [83/203  34% :: 0.722] Building C object lib/CMakeFiles/libcurl.dir/md5.c.o
[build] [84/203  34% :: 0.746] Building C object lib/CMakeFiles/libcurl.dir/llist.c.o
[build] [85/203  35% :: 0.747] Building C object lib/CMakeFiles/libcurl.dir/memdebug.c.o
[build] [86/203  35% :: 0.755] Building C object lib/CMakeFiles/libcurl.dir/openldap.c.o
[build] [87/203  36% :: 0.770] Building C object lib/CMakeFiles/libcurl.dir/netrc.c.o
[build] [88/203  36% :: 0.782] Building C object lib/CMakeFiles/libcurl.dir/nonblock.c.o
[build] [89/203  37% :: 0.799] Building C object lib/CMakeFiles/libcurl.dir/psl.c.o
[build] [90/203  37% :: 0.813] Building C object lib/CMakeFiles/libcurl.dir/http.c.o
[build] [91/203  38% :: 0.816] Building C object lib/CMakeFiles/libcurl.dir/parsedate.c.o
[build] [92/203  38% :: 0.825] Building C object lib/CMakeFiles/libcurl.dir/imap.c.o
[build] [93/203  39% :: 0.825] Building C object lib/CMakeFiles/libcurl.dir/mprintf.c.o
[build] [94/203  39% :: 0.842] Building C object lib/CMakeFiles/libcurl.dir/mqtt.c.o
[build] [95/203  40% :: 0.845] Building C object lib/CMakeFiles/libcurl.dir/pingpong.c.o
[build] [96/203  40% :: 0.899] Building C object lib/CMakeFiles/libcurl.dir/rename.c.o
[build] [97/203  41% :: 0.901] Building C object lib/CMakeFiles/libcurl.dir/sha256.c.o
[build] [98/203  41% :: 0.915] Building C object lib/CMakeFiles/libcurl.dir/progress.c.o
[build] [99/203  42% :: 0.923] Building C object lib/CMakeFiles/libcurl.dir/mime.c.o
[build] [100/203  42% :: 0.928] Building C object lib/CMakeFiles/libcurl.dir/select.c.o
[build] [101/203  43% :: 0.929] Building C object lib/CMakeFiles/libcurl.dir/slist.c.o
[build] [102/203  43% :: 0.951] Building C object lib/CMakeFiles/libcurl.dir/share.c.o
[build] [103/203  44% :: 0.968] Building C object lib/CMakeFiles/libcurl.dir/rand.c.o
[build] [104/203  44% :: 0.970] Building C object lib/CMakeFiles/libcurl.dir/rtsp.c.o
[build] [105/203  45% :: 0.976] Building C object lib/CMakeFiles/libcurl.dir/pop3.c.o
[build] [106/203  45% :: 0.983] Building C object lib/CMakeFiles/libcurl.dir/socks_sspi.c.o
[build] [107/203  46% :: 1.013] Building C object lib/CMakeFiles/libcurl.dir/socketpair.c.o
[build] [108/203  46% :: 1.015] Building C object lib/CMakeFiles/libcurl.dir/sendf.c.o
[build] [109/203  47% :: 1.030] Building C object lib/CMakeFiles/libcurl.dir/socks_gssapi.c.o
[build] [110/203  47% :: 1.039] Building C object lib/CMakeFiles/libcurl.dir/multi.c.o
[build] [111/203  48% :: 1.062] Building C object lib/CMakeFiles/libcurl.dir/splay.c.o
[build] [112/203  48% :: 1.066] Building C object lib/CMakeFiles/libcurl.dir/strdup.c.o
[build] [113/203  49% :: 1.068] Building C object lib/CMakeFiles/libcurl.dir/strtok.c.o
[build] [114/203  49% :: 1.081] Building C object lib/CMakeFiles/libcurl.dir/strcase.c.o
[build] [115/203  50% :: 1.090] Building C object lib/CMakeFiles/libcurl.dir/smb.c.o
[build] [116/203  50% :: 1.090] Building C object lib/CMakeFiles/libcurl.dir/socks.c.o
[build] [117/203  51% :: 1.091] Building C object lib/CMakeFiles/libcurl.dir/speedcheck.c.o
[build] [118/203  51% :: 1.091] Building C object lib/CMakeFiles/libcurl.dir/system_win32.c.o
[build] [119/203  52% :: 1.092] Building C object lib/CMakeFiles/libcurl.dir/strerror.c.o
[build] [120/203  52% :: 1.136] Building C object lib/CMakeFiles/libcurl.dir/timeval.c.o
[build] [121/203  53% :: 1.137] Building C object lib/CMakeFiles/libcurl.dir/strtoofft.c.o
[build] [122/203  53% :: 1.158] Building C object lib/CMakeFiles/libcurl.dir/smtp.c.o
[build] [123/203  54% :: 1.171] Building C object lib/CMakeFiles/libcurl.dir/warnless.c.o
[build] [124/203  54% :: 1.196] Building C object lib/CMakeFiles/libcurl.dir/vauth/cleartext.c.o
[build] [125/203  55% :: 1.202] Building C object lib/CMakeFiles/libcurl.dir/version_win32.c.o
[build] [126/203  55% :: 1.210] Building C object lib/CMakeFiles/libcurl.dir/version.c.o
[build] [127/203  56% :: 1.214] Building C object lib/CMakeFiles/libcurl.dir/setopt.c.o
[build] [128/203  56% :: 1.220] Building C object lib/CMakeFiles/libcurl.dir/tftp.c.o
[build] [129/203  57% :: 1.234] Building C object lib/CMakeFiles/libcurl.dir/wildcard.c.o
[build] [130/203  57% :: 1.248] Building C object lib/CMakeFiles/libcurl.dir/vauth/digest_sspi.c.o
[build] [131/203  58% :: 1.256] Building C object lib/CMakeFiles/libcurl.dir/telnet.c.o
[build] [132/203  58% :: 1.269] Building C object lib/CMakeFiles/libcurl.dir/transfer.c.o
[build] [133/203  59% :: 1.270] Building C object lib/CMakeFiles/libcurl.dir/vauth/cram.c.o
[build] [134/203  59% :: 1.280] Building C object lib/CMakeFiles/libcurl.dir/vauth/gsasl.c.o
[build] [135/203  60% :: 1.281] Building C object lib/CMakeFiles/libcurl.dir/vauth/ntlm_sspi.c.o
[build] [136/203  60% :: 1.288] Building C object lib/CMakeFiles/libcurl.dir/vauth/krb5_gssapi.c.o
[build] [137/203  61% :: 1.292] Building C object lib/CMakeFiles/libcurl.dir/vauth/krb5_sspi.c.o
[build] [138/203  61% :: 1.308] Building C object lib/CMakeFiles/libcurl.dir/urlapi.c.o
[build] [139/203  62% :: 1.315] Building C object lib/CMakeFiles/libcurl.dir/vauth/spnego_sspi.c.o
[build] [140/203  62% :: 1.333] Building C object lib/CMakeFiles/libcurl.dir/vauth/oauth2.c.o
[build] [141/203  63% :: 1.341] Building C object lib/CMakeFiles/libcurl.dir/vauth/vauth.c.o
[build] [142/203  63% :: 1.342] Building C object lib/CMakeFiles/libcurl.dir/vtls/gskit.c.o
[build] [143/203  64% :: 1.345] Building C object lib/CMakeFiles/libcurl.dir/vauth/spnego_gssapi.c.o
[build] [144/203  64% :: 1.358] Building C object lib/CMakeFiles/libcurl.dir/vauth/digest.c.o
[build] [145/203  65% :: 1.367] Building C object lib/CMakeFiles/libcurl.dir/vtls/bearssl.c.o
[build] [146/203  65% :: 1.374] Building C object lib/CMakeFiles/libcurl.dir/vtls/gtls.c.o
[build] [147/203  66% :: 1.375] Building C object lib/CMakeFiles/libcurl.dir/vtls/hostcheck.c.o
[build] [148/203  66% :: 1.375] Building C object lib/CMakeFiles/libcurl.dir/vtls/keylog.c.o
[build] [149/203  66% :: 1.383] Building C object lib/CMakeFiles/libcurl.dir/vtls/mbedtls_threadlock.c.o
[build] [150/203  67% :: 1.386] Building C object lib/CMakeFiles/libcurl.dir/vtls/mbedtls.c.o
[build] [151/203  67% :: 1.399] Building C object lib/CMakeFiles/libcurl.dir/vtls/nss.c.o
[build] [152/203  68% :: 1.410] Building C object lib/CMakeFiles/libcurl.dir/vtls/schannel_verify.c.o
[build] [153/203  68% :: 1.416] Building C object lib/CMakeFiles/libcurl.dir/vauth/ntlm.c.o
[build] [154/203  69% :: 1.430] Building C object lib/CMakeFiles/libcurl.dir/vtls/schannel.c.o
[build] [154/203  69% :: 1.433] Building C object lib/CMakeFiles/libcurl.dir/vtls/rustls.c.o
[build] [154/203  70% :: 1.444] Building C object lib/CMakeFiles/libcurl.dir/vquic/ngtcp2.c.o
[build] [154/203  70% :: 1.455] Building C object lib/CMakeFiles/libcurl.dir/vquic/quiche.c.o
[build] [154/203  71% :: 1.456] Building C object lib/CMakeFiles/libcurl.dir/vtls/wolfssl.c.o
[build] [154/203  71% :: 1.461] Building C object lib/CMakeFiles/libcurl.dir/vtls/x509asn1.c.o
[build] [154/203  72% :: 1.462] Building C object lib/CMakeFiles/libcurl.dir/url.c.o
[build] [154/203  72% :: 1.467] Building C object lib/CMakeFiles/libcurl.dir/vssh/libssh.c.o
[build] [154/203  73% :: 1.472] Building C object lib/CMakeFiles/libcurl.dir/vssh/wolfssh.c.o
[build] [154/203  73% :: 1.472] Building C object lib/CMakeFiles/libcurl.dir/vssh/libssh2.c.o
[build] [154/203  74% :: 1.484] Building C object lib/CMakeFiles/libcurl.dir/vquic/vquic.c.o
[build] [154/203  74% :: 1.486] Building C object lib/CMakeFiles/libcurl.dir/vtls/sectransp.c.o
[build] [154/203  75% :: 1.530] Building C object lib/CMakeFiles/libcurl.dir/vtls/vtls.c.o
[build] [154/203  75% :: 1.537] Building C object lib/CMakeFiles/libcurl.dir/vtls/openssl.c.o
[build] FAILED: lib/CMakeFiles/libcurl.dir/vtls/openssl.c.o 
[build] /bin/x86_64-linux-gnu-gcc-9 -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -DHAVE_CONFIG_H -DOPENSSL_SUPPRESS_DEPRECATED -Dlibcurl_EXPORTS -I../include -Ilib/../include -I../lib/.. -I../lib/../include -Ilib/.. -I../lib -Ilib -pedantic -Wall -W -Wpointer-arith -Wwrite-strings -Wunused -Wshadow -Winline -Wnested-externs -Wmissing-declarations -Wmissing-prototypes -Wfloat-equal -Wsign-compare -Wundef -Wendif-labels -Wstrict-prototypes -Wdeclaration-after-statement -Wstrict-aliasing=3 -Wcast-align -Wtype-limits -Wold-style-declaration -Wmissing-parameter-type -Wempty-body -Wclobbered -Wignored-qualifiers -Wconversion -Wvla -Wdouble-promotion -Wno-long-long -Wno-multichar -Wno-format-nonliteral -Wno-sign-conversion -Wno-system-headers -g -fPIC   -fvisibility=hidden -MD -MT lib/CMakeFiles/libcurl.dir/vtls/openssl.c.o -MF lib/CMakeFiles/libcurl.dir/vtls/openssl.c.o.d -o lib/CMakeFiles/libcurl.dir/vtls/openssl.c.o   -c ../lib/vtls/openssl.c
[build] ../lib/vtls/openssl.c: In function ‘ossl_strerror’:
[build] ../lib/vtls/openssl.c:388:22: warning: conversion from ‘long unsigned int’ to ‘uint32_t’ {aka ‘unsigned int’} may change value [-Wconversion]
[build]   388 |   ERR_error_string_n(error, buf, size);
[build]       |                      ^~~~~
[build] ../lib/vtls/openssl.c: In function ‘Curl_ossl_verifyhost’:
[build] ../lib/vtls/openssl.c:1733:15: warning: conversion from ‘size_t’ {aka ‘long unsigned int’} to ‘int’ may change value [-Wconversion]
[build]  1733 |     numalts = sk_GENERAL_NAME_num(altnames);
[build]       |               ^~~~~~~~~~~~~~~~~~~
[build] ../lib/vtls/openssl.c: In function ‘set_ssl_version_min_max’:
[build] ../lib/vtls/openssl.c:2387:44: warning: conversion from ‘long int’ to ‘uint16_t’ {aka ‘short unsigned int’} may change value [-Wconversion]
[build]  2387 |     if(!SSL_CTX_set_min_proto_version(ctx, ossl_ssl_version_min)) {
[build]       |                                            ^~~~~~~~~~~~~~~~~~~~
[build] ../lib/vtls/openssl.c:2422:42: warning: conversion from ‘long int’ to ‘uint16_t’ {aka ‘short unsigned int’} may change value [-Wconversion]
[build]  2422 |   if(!SSL_CTX_set_max_proto_version(ctx, ossl_ssl_version_max)) {
[build]       |                                          ^~~~~~~~~~~~~~~~~~~~
[build] ../lib/vtls/openssl.c: In function ‘ossl_connect_step1’:
[build] ../lib/vtls/openssl.c:2832:37: warning: conversion from ‘ctx_option_t’ {aka ‘long int’} to ‘uint32_t’ {aka ‘unsigned int’} may change value [-Wconversion]
[build]  2832 |   SSL_CTX_set_options(backend->ctx, ctx_options);
[build]       |                                     ^~~~~~~~~~~
[build] ../lib/vtls/openssl.c:2900:11: warning: implicit declaration of function ‘SSL_CTX_set_ciphersuites’; did you mean ‘SSL_CTX_set_cipher_list’? [-Wimplicit-function-declaration]
[build]  2900 |       if(!SSL_CTX_set_ciphersuites(backend->ctx, ciphers13)) {
[build]       |           ^~~~~~~~~~~~~~~~~~~~~~~~
[build]       |           SSL_CTX_set_cipher_list
[build] ../lib/vtls/openssl.c:2900:11: warning: nested extern declaration of ‘SSL_CTX_set_ciphersuites’ [-Wnested-externs]
[build] ../lib/vtls/openssl.c:2911:3: warning: implicit declaration of function ‘SSL_CTX_set_post_handshake_auth’ [-Wimplicit-function-declaration]
[build]  2911 |   SSL_CTX_set_post_handshake_auth(backend->ctx, 1);
[build]       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build] ../lib/vtls/openssl.c:2911:3: warning: nested extern declaration of ‘SSL_CTX_set_post_handshake_auth’ [-Wnested-externs]
[build] ../lib/vtls/openssl.c: In function ‘ossl_connect_step2’:
[build] ../lib/vtls/openssl.c:3405:26: error: ‘SSL_R_TLSV13_ALERT_CERTIFICATE_REQUIRED’ undeclared (first use in this function); did you mean ‘SSL_R_TLSV1_ALERT_CERTIFICATE_REQUIRED’?
[build]  3405 |               (reason == SSL_R_TLSV13_ALERT_CERTIFICATE_REQUIRED)) {
[build]       |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |                          SSL_R_TLSV1_ALERT_CERTIFICATE_REQUIRED
[build] ../lib/vtls/openssl.c:3405:26: note: each undeclared identifier is reported only once for each function it appears in
[build] ../lib/vtls/openssl.c: In function ‘get_cert_chain’:
[build] ../lib/vtls/openssl.c:3597:14: warning: conversion from ‘size_t’ {aka ‘long unsigned int’} to ‘numcert_t’ {aka ‘int’} may change value [-Wconversion]
[build]  3597 |   numcerts = sk_X509_num(sk);
[build]       |              ^~~~~~~~~~~
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

### curl/libcurl version

[curl -V output]

I am building from the HEAD of the repository.

### operating system

```
Linux ThinkPad-X1-Carbon-7th 5.13.0-35-generic #40~20.04.1-Ubuntu SMP Mon Mar 7 09:18:32 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by bagder** [comment]
What OpenSSL version are you building against?

**Comment by cc-michaelu** [comment]
> What OpenSSL version are you building against?

Hello Daniel,

**Comment by cc-michaelu** [comment]
I just tried commenting out the code where this undeclared symbol is used, but now I see linkage errors (which I also saw trying to build older tags of this project):

```
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `BIO_read_filename'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `BIO_reset'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set_session_cache_mode'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set_min_proto_version'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `sk_value'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set1_curves_list'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `EVP_MD_CTX_create'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set_max_proto_version'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set_mode'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_add0_chain_cert'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_add_extra_chain_cert'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `BIO_get_mem_data'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `EVP_MD_CTX_destroy'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_clear_chain_certs'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `sk_num'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_set_tlsext_host_name'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_CTX_set_msg_callback_arg'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `BIO_get_mem_ptr'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `sk_pop'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `SSL_get_ex_new_index'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `OPENSSL_malloc'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `OPENSSL_free'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `BIO_set_ssl'
[build] /usr/bin/ld: lib/libcurl-d.so: undefined reference to `sk_pop_free_ex'
[build] collect2: error: ld returned 1 exit status
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

**Comment by cc-michaelu** [comment]
I dont have time today to investigate, but I was thinking of grabbing the openssl projects code and building it from source, installing it to a temporary directory and building curl pointing to the custom ssl build.... But I'll have to put this off till later as I need to get to work on other issues.

**Comment by cc-michaelu** [comment]
@bagder I have tried building on another laptop running the same Ubuntu 20.04 and it builds with no errors.

So I think this should be closed because the problem is not with the project, rather the build environment on my main development machine. I dont have time to debug the problem and as I now can do what I need on my laptop I dont need this resolved.

Thank you

**Comment by JanneKiiskila** [comment]
This has actually broken somehow, I'm getting a similar issue and it used to work just fine with an earlier version of libcurl - version 7.76.0 at least for sure. @jay

**Comment by JanneKiiskila** [comment]
Works also with version 7.80.0.

**Comment by JanneKiiskila** [comment]
Breaks at 7.81.0.

```
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY) (found
  version "1.1.1f")
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindOpenSSL.cmake:447 (find_package_handle_standard_args)
  /home/jannek/mbed/mbed-cloud-client-example-internal/pal-platform/Middleware/curl/curl/CMakeLists.txt:443 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/jannek/mbed/mbed-cloud-client-example-internal/__x86_x64_NativeLinux_mbedtls/CMakeFiles/CMakeOutput.log".
See also "/home/jannek/mbed/mbed-cloud-client-example-internal/__x86_x64_NativeLinux_mbedtls/CMakeFiles/CMakeError.log".
```

**Comment by bagder** [comment]
@JanneKiiskila that looks like a different problem though

**Comment by bagder** [comment]
I just built curl from the git repo successfully on ubuntu 20.04 first using configure then cmake, both with OpenSSL 1.1.1f

**Comment by JanneKiiskila** [comment]
Yep, seems the changes related to usage of SSL-stacks has changed between 7.80 and 7.81, which caused my issues.

**Comment by bagder** [comment]
curl's cmake support is always lagging behind...

**Comment by jay** [comment]
> We were using `CMAKE_USE_OPENSSL` in our `CMakefiles.txt`, I changed that to `CURL_USE_OPENSSL` and it started working.
> 
> What's a bit strange is that why didn't the curl's CMakefiles.txt error message guidance tell me this directly, as it actually has handling for this [in place already](https://github.com/curl/curl/blob/master/CMakeLists.txt#L348-L356)?

Ref: #8052
Ref: 0969805

Starting in 7.81.0 it should error when those symbols are used. I can't reproduce with master 5e9a703 2022-05-02. I tried `-DCMAKE_USE_OPENSSL=ON` on both an existing valid configuration and from scratch. Either way causes an error. If you find a way to reproduce with master then please open a new issue.
